staging/src/k8s.io/apimachinery/pkg/api/errors/errors_test.go

```diff
@@ -220,3 +220,39 @@ func TestFromObject(t *testing.T) {
 		}
 	}
 }
+
+func TestReasonForErrorSupportsWrappedErrors(t *testing.T) {
+	testCases := []struct {
+		name                string
+		err                 error
+		expectedKnownReason bool
+	}{
+		{
+			name:                "Direct match",
+			err:                 &StatusError{ErrStatus: metav1.Status{Reason: metav1.StatusReasonUnauthorized}},
+			expectedKnownReason: true,
+		},
+		{
+			name: "No match",
+			err:  errors.New("some other error"),
+		},
+		{
+			name:                "Nested match",
+			err:                 fmt.Errorf("wrapping: %w", fmt.Errorf("some more: %w", &StatusError{ErrStatus: metav1.Status{Reason: metav1.StatusReasonAlreadyExists}})),
+			expectedKnownReason: true,
+		},
+		{
+			name: "Nested, no match",
+			err:  fmt.Errorf("wrapping: %w", fmt.Errorf("some more: %w", errors.New("hello"))),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			result := ReasonForError(tc.err)
+			if knownReason := (result != metav1.StatusReasonUnknown); knownReason != tc.expectedKnownReason {
```

Comment:
If you do match the type of status, then you would be able to remove the hardcode here, and put `StatusReasonUnknown` for each test that is unknown.